src/cpp/session/SessionSuspend.cpp

```diff
@@ -71,6 +71,13 @@ void setSuspendedFromTimeout(bool suspended)
    
 bool suspendSession(bool force, int status)
 {
+   if (!options().timeoutSuspend())
```

Comment:
I think this will make admin commands such as `rstudio-server suspend-all` quit sessions rather than suspending them, which is probably not what we want, and it may also break suspend-and-restart actions in the UI. We probably need to detect the timeout at a higher level and issue a quit command to the session rather than subverting the suspension logic.